pkg/api/pod/warnings.go

```diff
@@ -141,7 +141,7 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 			warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.11, non-functional in v1.16+", fieldPath.Child("spec", "volumes").Index(i).Child("photonPersistentDisk")))
 		}
 		if v.GitRepo != nil {
-			warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.11", fieldPath.Child("spec", "volumes").Index(i).Child("gitRepo")))
+			warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.11, non-functional in v1.33+", fieldPath.Child("spec", "volumes").Index(i).Child("gitRepo")))
```

Comment:
Would it make sense to be more explicit: `non-functional in v1.33+` -> `to be disabled/removed in v1.33+`?

Reply:
the disablement is still conditional / gated, so only append this bit if the gate is disabled, and don't assert removal in future releases yet

Reply:
Gated the response on feature-gate enablement.